Sources/Rendering/OpenGL/Texture/index.js

```diff
@@ -649,9 +655,20 @@ function vtkOpenGLTexture(publicAPI, model) {
 
     // if the opengl data type is half float
     // then the data array must be u16
-    const halfFloat = model.context.getExtension('OES_texture_half_float');
-    if (halfFloat && model.openGLDataType === halfFloat.HALF_FLOAT_OES) {
-      const pixCount = model.width * model.height * model.components;
+    const halfFloatExt = model.context.getExtension('OES_texture_half_float');
+    const halfFloat = model.openGLRenderWindow.getWebgl2()
+      ? model.openGLDataType === model.context.HALF_FLOAT
+      : halfFloatExt && model.openGLDataType === halfFloatExt.HALF_FLOAT_OES;
+
+    if (
+      halfFloat &&
+      (dataType === VtkDataTypes.UNSIGNED_SHORT ||
+        dataType === VtkDataTypes.SHORT)
+    ) {
```

Comment:
I think this will miss the case on OpenGL1 where we have FLOAT data but are stuck rendering it as half float. I think you can just check on if (halfFloat) here as if the opengl data type is half float then we know we need to be doing a conversion. Can you double check my logic here in case I'm missing something?

Reply:
Yes, you are right, it will miss it. I also think `if (halfFLoat)` covers it all